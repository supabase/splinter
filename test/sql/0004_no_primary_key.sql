begin;
    set local search_path = '';

    -- No issues
    select * from lint."0004_no_primary_key";

    -- Table with a primary key
    create table public.foo (
        id int primary key
    );

    -- Table with a primary key
    create table public.bar (
        id int
    );

    -- Only the "bar" table is listed
    select * from lint."0004_no_primary_key";

rollback;
