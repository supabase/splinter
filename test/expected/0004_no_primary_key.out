begin;
    -- No issues
    select * from lint."0004_no_primary_key";
 name | level | facing | description | detail | remediation | metadata | cache_key 
------+-------+--------+-------------+--------+-------------+----------+-----------
(0 rows)

    -- Table with a primary key
    create table public.foo (
        id int primary key
    );
    -- Table with a primary key
    create table public.bar (
        id int
    );
    -- Only the "bar" table is listed
    select * from lint."0004_no_primary_key";
      name      | level |  facing  |                                                        description                                                         |                      detail                      |                                    remediation                                     |                       metadata                       |         cache_key         
----------------+-------+----------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------+------------------------------------------------------------------------------------+------------------------------------------------------+---------------------------
 no_primary_key | INFO  | EXTERNAL | Detects if a table does not have a primary key. Tables without a primary key can be inefficient to interact with at scale. | Table \`public.bar\` does not have a primary key | https://supabase.com/docs/guides/database/database-linter?lint=0004_no_primary_key | {"name": "bar", "type": "table", "schema": "public"} | no_primary_key_public_bar
(1 row)

rollback;
